### PR TITLE
refactor: redo column types with a new column_types field

### PR DESCRIPTION
**Pull request by odinuv**
https://keboola.atlassian.net/browse/PS-717

plus fixnuty bugisek z predchoziho merge https://github.com/keboola/input-mapping/pull/55/commits/698735dd972b08933d86f619941e8f7bf2ff6a99

plus jsem tam pridal kontrolu a test na to, kdyz se spusti komponenta s s3 mappingem na abs backendu a obracene (koncilo to hlaskou undefined index)

### CODE REVIEW

**Comment by odinuv** [comment]
update runneru https://github.com/keboola/docker-bundle/pull/484

File: Tests/Reader/Options/InputTableOptionsTest.php [hunk 1]
```diff
@@ -181,17 +182,45 @@ public function testGetLoadOptionsExtendedColumns()
         ], $definition->getStorageApiLoadOptions(new InputTableStateList([])));
     }
 
-    public function testInvalidColumns()
+    public function testInvalidColumnsMissing()
     {
         self::expectException(InvalidInputException::class);
         self::expectExceptionMessage(
-            'Columns must be either specified as array of strings or as array of objects with type, but not both'
+            'Both columns and column_types are specified, columns field contains surplus columns: col1.'
```
Comment (MiroCillik) [inline Tests/Reader/Options/InputTableOptionsTest.php:189]: Tu by som mozno tie nazvy atributov "columns" a "column_types" dal do nejakych zatvoriek, nech je jasne, ze su to atributy.

Comment (MiroCillik) [reply Tests/Reader/Options/InputTableOptionsTest.php:189]: Tym tu myslim tu https://github.com/keboola/input-mapping/pull/55/files#diff-b7b5320ee9976c0ce0148f8c3714803bR42 :)

Comment (odinuv) [reply Tests/Reader/Options/InputTableOptionsTest.php:189]: > Myslim, ze dobry, v docker runneri to testami tiez preslo, takze by to malo byt OK.
> 
> Len pre istotu sa spytam. Teraz teda nemoze nastat situacia, ze je v `columns` ten zlozitejsi format?
> Vzdy to bude len jednoduche pole nazvov stlpcov?
> 
> Len by som pridal do tych diff chybovych hlasok tie uvodzovky, aby bolo jasne kedy myslis "columns" ako atribut a kedy ako normalne slovo.

Jo ten slozitejsi columns format jsme nereleasli, takze to to nikde neni a tady https://github.com/keboola/input-mapping/blob/698735dd972b08933d86f619941e8f7bf2ff6a99/Configuration/Table.php#L39 zustava pole scalaru

akorat jsou vlastne ted dve moznosti jak omezit sloupce tabulky - bud to zadat v columns, nebo to zadat v column_types (coz naplni automaticky i columns), pripadne zadat oboji, ale pak to musi byt stejny sloupce

**Pull request merged**